### PR TITLE
Integrate Google OAuth2 login on /auth/google/

### PR DESCRIPTION
**Pull request by kevinschoonover**

### CODE REVIEW

File: src/middleware/auth.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  const user = await Account.findOne({id: userID});
+  if (user){
+    done(null, user);
+  } else {
+    done(null, false);
+  }
+});
+
+
+// TODO: remove hard-coded userProfileURL
```
Comment (ClayMav) [inline src/middleware/auth.ts:21]: TODO in code, should it be?

File: src/middleware/auth.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+// TODO: change callbackURL to support standard website
```
Comment (ClayMav) [inline src/middleware/auth.ts:22]: TODO in code, should it be?

**Review by ClayMav — DISMISSED** [review]
Well, I tried to pull and run this myself and it would not run. Lets talk about it.

File: src/middleware/auth.ts (hunk 1 above)
Comment (kevinschoonover) [reply src/middleware/auth.ts:21]: The change it out of the scope of the specific PR and requires a lot more work

File: src/middleware/auth.ts (hunk 2 above)
Comment (kevinschoonover) [reply src/middleware/auth.ts:22]: These changes are important for developing future features, but not production ready. I am willing to forgo the temporary problems for the productivity.

**Comment by kevinschoonover** [comment]
@ClayMav Did you run the code using Google OAUTH2 keys? Look in the readme for more info.

**Comment by kevinschoonover** [comment]
@ClayMav You can go ahead and re-review it.

File: src/middleware/auth.test.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+      Account.mockImplementation(() => {
+        return {
+          save() {
+            this.id = "test";
+            user = this;
+          },
+        };
+      });
+      Account.findOne = jest.fn()
+        .mockReturnValueOnce(undefined)
```
Comment (rds6y9) [inline src/middleware/auth.test.ts]: Do we need both mockReturnValueOnce() and mockReturnValue() if they both are the same return?

**Review by rds6y9 — APPROVED** [review]
Only note was about what might be a redundancy. Other than that, looks good.

**Pull request merged**